src/include/duckdb/main/database.hpp

```diff
@@ -79,6 +84,7 @@ class DatabaseInstance : public enable_shared_from_this<DatabaseInstance> {
 	unique_ptr<ConnectionManager> connection_manager;
 	unordered_set<string> loaded_extensions;
 	unordered_map<string, ExtensionInstallInfo> loaded_extensions_data;
+	unordered_map<string, ExtensionLoadedInfo> loaded_extensions_extra_data;
```

Comment:
Looks like we now have three maps/sets that all have the exact same keys, can we merge `loaded_extensions`, `loaded_extensions_data` and `loaded_extensions_extra_data`?

Reply:
I think that makes sense, sure.
One thing is that currently it's not a given that the 3 maps have the same cardinality / but having a map string to a struct like:
```
{
    bool isLoaded;
    unique_ptr<ExtensionInstallInfo> install_info;
    unique_ptr<ExtensionLoadedInfo> loaded_info;
}
```
plus having iterators check for validity is cleaner.